tests/test_smart_ptr.cpp

```diff
@@ -353,6 +363,15 @@ TEST_SUBMODULE(smart_ptr, m) {
     m.def("print_myobject3_3", [](const std::shared_ptr<MyObject3> &obj) { py::print(obj->toString()); });
     m.def("print_myobject3_4", [](const std::shared_ptr<MyObject3> *obj) { py::print((*obj)->toString()); });
 
+    py::class_<MyObject3a>(m, "MyObject3a");
+    m.def("make_myobject3_1", []() { return new MyObject3a(8); });
+    m.def("make_myobject3_2", []() { return std::make_shared<MyObject3a>(9); });
+    m.def("print_myobject3a_1", [](const MyObject3a *obj) { py::print(obj->toString()); });
+    m.def("print_myobject3a_2", [](std::shared_ptr<MyObject3a> obj) { py::print(obj->toString()); });
+    m.def("print_myobject3a_3", [](const std::shared_ptr<MyObject3a> &obj) { py::print(obj->toString()); });
```

Comment:
All these functions are not yet used, are they?

Reply:
FWIW, I copied this also under my WIP #3023, but at the moment I'm thinking this block doesn't add anything in the smart_holder context. Focusing this PR narrowly on the `copy` return value policy question seems best to me.

Reply:
Right, these were just copy-paste artefacts, I'll clean it up when moving the test to a separate file.